### PR TITLE
[EventHubs] Stress Test

### PR DESCRIPTION
**Pull request by yunhaoling**
Issue: #9038

- [x] Stress for producer, sync/async
- [x] Stress for receiver, sync/async
- [x] Stress for receiver with checkpoint manager, sync/async

- [x] adding more configurations
- [x] parallel receive/send
- [x] process monitor

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+from process_monitor import ProcessMonitor
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--link_credit", default=3000, type=int)
+parser.add_argument("--output_interval", type=float, default=1000)
+parser.add_argument("--duration", help="Duration in seconds of the test", type=int, default=30)
+parser.add_argument("--consumer_group", help="Consumer group name", default="$default")
+parser.add_argument("--auth_timeout", help="Authorization Timeout", type=float, default=60)
+parser.add_argument("--offset", help="Starting offset", default="-1")
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: Change to event_position? Not necessary to be an offset

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+parser.add_argument("--address", help="Address URI to the EventHub entity")
+parser.add_argument("--sas_policy", help="Name of the shared access policy to authenticate with")
+parser.add_argument("--sas_key", help="Shared access key")
+parser.add_argument(
+    "--transport_type",
+    help="Transport type, 0 means AMQP, 1 means AMQP over WebSocket",
+    type=int,
+    default=0
+)
+parser.add_argument("--parallel_recv_cnt", help="Number of parallelling receiving", type=int)
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: What is this?

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+parser.add_argument("--proxy_hostname", type=str)
+parser.add_argument("--proxy_port", type=str)
+parser.add_argument("--proxy_username", type=str)
+parser.add_argument("--proxy_password", type=str)
+parser.add_argument("--aad_client_id", help="AAD client id")
+parser.add_argument("--aad_secret", help="AAD secret")
+parser.add_argument("--aad_tenant_id", help="AAD tenant id")
+parser.add_argument("--storage_conn_str", help="conn str of storage blob to store ownership and checkpoint data")
+parser.add_argument("--storage_container_name", help="storage container name to store ownership and checkpoint data")
+parser.add_argument("--uamqp_debug", help="uamqp logging enable", action="store_true")
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: change to uamqp_logging_enable to have consistent names?

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+            load_balancing_interval=args.load_balancing_interval,
+            auth_timeout=args.auth_timeout,
+            http_proxy=http_proxy,
+            transport_type=transport_type,
+            logging_enable=args.uamqp_debug
+        )
+
+    return client
+
+    return client
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: Unreachable code

**Comment by YijunXieMS** [comment]
LGTM

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py (hunk 4 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: removed

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py (hunk 3 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: updated

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py (hunk 2 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: This is for creating multiple clients to receive simultaneously

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/stress/azure_eventhub_consumer_stress_async.py]: added new args, now becoming
- starting_offset
- starting_sequence_number
- starting_datetime (str should be format of YYYY-mm-dd HH:mm:ss)

File: sdk/eventhub/azure-eventhub/stress/azure_eventhub_producer_stress.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            start_time = time.perf_counter()
+            while self.running and time.time() < deadline:
+                try:
+                    processed = await test_method(worker, self.args, logger)
+                    total_processed += processed
+                    iter_processed += processed
+                    if iter_processed >= self.args.output_interval:
+                        time_elapsed = time.perf_counter() - start_time
+                        logger.info(
+                            "Total sent: %r, Time elapsed: %r, Speed: %r/s",
```
[the remark this answers is not in the document]
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/stress/azure_eventhub_producer_stress.py]: added speed metric for current iteration

**Pull request merged**